components/studio/bin/hab-studio.ps1

```diff
@@ -315,28 +325,31 @@ function Enter-Studio {
     New-PSDrive -Name "Habitat" -PSProvider FileSystem -Root $env:HAB_STUDIO_ENTER_ROOT | Out-Null
     mkdir $env:HAB_STUDIO_ENTER_ROOT\hab\sup\default -Force | Out-Null
     
-    $pr = New-Object System.Diagnostics.Process
-    $pr.StartInfo.UseShellExecute = $false
-    $pr.StartInfo.CreateNoWindow = $true
-    $pr.StartInfo.RedirectStandardOutput = $true
-    $pr.StartInfo.RedirectStandardError = $true
-    $pr.StartInfo.FileName = "hab.exe"
-    $pr.StartInfo.Arguments = "sup run"
-    Register-ObjectEvent -InputObject $pr -EventName OutputDataReceived -action {
-      $Event.SourceEventArgs.Data | Out-File $env:HAB_STUDIO_ENTER_ROOT\hab\sup\default\out.log -Append -Encoding ascii
-    } | Out-Null
-    Register-ObjectEvent -InputObject $pr -EventName ErrorDataReceived -action {
-      $Event.SourceEventArgs.Data | Out-File $env:HAB_STUDIO_ENTER_ROOT\hab\sup\default\out.log -Append -Encoding ascii
-    } | Out-Null
-    $pr.start() | Out-Null
-    $pr.BeginErrorReadLine()
-    
-    # As an industry, this should make us all feel bad
-    # without this stdin never seems to come back and the PS prompt
-    # fails to appear and things look like they are hung
-    Start-Sleep -Milliseconds 100
-
-    $pr.BeginOutputReadLine()
+    $habSvc = Get-Service Habitat -ErrorAction SilentlyContinue
+    if(!$habSvc -or ($habSvc.Status -eq "Stopped")) {
+      $pr = New-Object System.Diagnostics.Process
+      $pr.StartInfo.UseShellExecute = $false
+      $pr.StartInfo.CreateNoWindow = $true
+      $pr.StartInfo.RedirectStandardOutput = $true
+      $pr.StartInfo.RedirectStandardError = $true
+      $pr.StartInfo.FileName = "hab.exe"
+      $pr.StartInfo.Arguments = "sup run"
+      Register-ObjectEvent -InputObject $pr -EventName OutputDataReceived -action {
+        $Event.SourceEventArgs.Data | Out-File $env:HAB_STUDIO_ENTER_ROOT\hab\sup\default\out.log -Append -Encoding ascii
+      } | Out-Null
```

Comment:
That's pretty cool IO redirection--this way it logs to `out.log` but doesn't also log to `stdout`?

Reply:
yup `out-null` == `/dev/null`